trade-finance-manager-ui/server/controllers/utilisation-reports/update-utilisation-report-status/index.ts

```diff
@@ -67,12 +69,27 @@ const getReportWithStatus = (reportIdentifier: ReportIdentifier, formButton: str
   }
 };
 
-export const updateUtilisationReportStatus = async (req: Request, res: Response) => {
+type UpdateUtilisationReportStatusRequestBody = {
+  _csrf: string;
+  'form-button': string;
+  'submission-month': string;
+} & {
+  [key: string]: 'on'; // all checkboxes in payload have value 'on'
+};
+
+interface UpdateUtilisationReportStatusRequest extends Request {
+  body: UpdateUtilisationReportStatusRequestBody;
+}
```

Comment:
This feels like a bit of a hacky way of setting the type of `body` rather than using the generic parameters of the `Request` type. It could be that the `ReqBody` type parameter is used to set types elsewhere within `Request` (I'm not sure if it is, but the point stands).

Reply:
I actually think this is the correct way of doing it rather than using the generic. `interface` allows us to extend the type whilst preserving the the original details (primarily required for the `query` type definition), whereas `Request<{}, {}, [Body]>` causes two errors:
- firstly, you can't use `{}` because of some eslint rule
- secondly, changing to `object` works for the type of `req` in this function the same was as `interface`, but when you try to pass that request down to `getUtilisationReports`, you get a new error saying that `"Type 'object' is not assignable to type 'ParamsDictionary'."`

Extending the `interface` instead resolves these issues as only the type we actually modify in `Request` is updated and the rest are kept at their original states

Reply:
Yeah, I get that, but the fact that we're not using the generics provided by the `Request` type means that if `ReqBody` (the third param) is used anywhere other than on the `body` property, then this way means it will be ignored. I see other people are using it, but that doesn't convince me this is good practice.

It is a pain though to have to provide the defaults for the other properties, so I can see why it would be done this way. It's not enough of a concern for me to hold this up and look into further, so happy to go with what you've done.